### ISSUE TITLE
Implement certain git status operations without spawning git

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode#64234]
Like finding HEAD and all refs, etc.

cc @rebornix

**Comment by lszomoru** [comment]
Here is the set of `git.exe` calls while updating the repository model. Some execute sequentially while some in parallel:
```
[2022-10-03T11:37:24.163Z] > git status -z -uall [679ms]
[2022-10-03T11:37:24.176Z] > git symbolic-ref --short HEAD [11ms]
[2022-10-03T11:37:24.217Z] > git for-each-ref --format=%(refname)%00%(upstream:short)%00%(objectname)%00%(upstream:track)%00%(upstream:remotename)%00%(upstream:remoteref) refs/heads/main refs/remotes/main [39ms]
[2022-10-03T11:37:24.232Z] > git remote --verbose [12ms]
[2022-10-03T11:37:24.855Z] > git for-each-ref --sort -committerdate --format %(refname) %(objectname) %(*objectname) [637ms]
[2022-10-03T11:37:24.880Z] > git config --get commit.template [11ms]
```

**Comment by lszomoru** [comment]
(2) has been implemented. The rest I will look at next milestone.

**Comment by lszomoru** [comment]
(4) has been implemented as well.

**Comment by lszomoru** [comment]
The rest is currently out of scope.

**Comment by lszomoru** [comment]
Verification steps:
* Open a folder/workspace that contains a git repository with multiple remotes
* Open the "Source Control" viewlet and click the refresh button multiple times
  * Open the git output window, and confirm that there are no executions of `git remote --verbose`
* Open the Command palette and invoke the "Git: Fetch" command
  * Confirm that the quick pick contains all existing remotes
